### ISSUE TITLE
[Feature Request] PVR Client Addon for Android's native TIF (TV Input Framework)?

### ISSUE THREAD

**Issue by Hedda** [koying/SPMC#75]
Appently from Android 5.0 (API level 21) the OS have an integrated TV Input Framework (TIF) with built-in tuner input support.

@koying Wondering if it would be possible (and a good idea) to make a PVR Client Addon for SPMC (and Kodi for Android) with native support for Android TIF?

Or would a better architectual design be to add support for Android TIF into a PVR backend like Tvheadend instead and port that to run as a service on Android?

http://source.android.com/devices/tv/
http://developer.android.com/training/tv/tif/index.html
https://developer.android.com/reference/android/media/tv/package-summary.html
https://github.com/googlesamples/androidtv-sample-inputs

Point is it would be nice to have a PVR Client Addon for Kodi's PVR API that can access Android TIF and TV tuners to watch live televsion received from a broadcast signal via antenna or cable/satellite.

There are now loads of Android boxes out there that have integrated TV tuners, like example WeTek Play / Play2 series, but there is today not possible to access local TV tuners (built-in or connected directly via USB adapter).

Just imagine how awesome it would be for each Android box with SPMC/Kodi to be able to function as a standalone set-top box that can receieve local broadcast television directly to not have to rely on an external PVR server or IPTV.

**Comment by donaldsmith2060** [comment]
Check out how dvblink does this and the news that in android n you'll be able to schedule and view recordings. They have a beta backend for the tif framework.

**Comment by Fleker** [comment]
Yeah having support for Live Channels would be fantastic. Would it make sense to have it as a PVR plugin? It would be great for many of the video plugins to be able to automatically play as a channel in Live Channels.

You can look at my library http://github.com/fleker/ChannelSurfer which removes a lot of the TV-input framework boilerplate.

Adding in DVR support on Android N would also be great down the line.

**Comment by qzem** [comment]
Is there anything new on this issue. I am also searching for a way to use native DVB-C receiver on Philips Android TV with Kodi

**Comment by sammarbella** [comment]
Waiting for Oreo...

**Comment by Hedda** [comment]
What's new in Android TIF APIs on Android Oreo? Android 8.0 = API level 26, 8.1 =  Android API level 27

https://source.android.com/devices/tv/		

https://developer.android.com/training/tv/tif/tvinput.html

**Comment by Fleker** [comment]
AFAIK the TIF API did not change in API 26 or 27.
